spec/lib/common/model-spec.js

```diff
@@ -341,57 +340,261 @@ describe('Model', function () {
                 expect(removed.list).to.deep.equal(['item1', 'item2']);
             });
 
+            it('should return the same doc with blank value', function () {
+                return waterline.testobjects.removeListItemsById(record.id, {
+                    list: []
+                }).then(function(removed_){
+                        expect(removed.list).to.deep.equal(removed_.list);
+                });
+            });
+        });
+
+        describe('remove list items by bad id with removeListItemsById()',
+            function () {
+            it('should reject with a not found error', function () {
+                return waterline.testobjects.removeListItemsByIdentifier('invalid id', ["item1"])
+                .should.be.rejectedWith(Errors.NotFoundError);
+            });
+        });
+
+        describe('remove list items by bad value with removeListItemsByIdentifier()',
+            function () {
+            it('should reject with a mongo error', function () {
+                return waterline.testobjects.removeListItemsByIdentifier('qwe', '')
+                .should.be.rejectedWith(Errors.MongoError);
+            });
         });
 
         describe('remove list items by identifier with removeListItemsByIdentifier()', function () {
-            var removed;
+            var value = { list: ['item1', 'item2', 'item3'] };
+
+            before('set up mocks', function () {
+               waterline.testobjects.removeListItemsById = sinon.stub().returns(bluebird.resolve());
+            });
+
+            it('should called removeListItemsById', function(){
+                return waterline.testobjects.removeListItemsByIdentifier("qwe", value)
+                .tap(function(){
+                    expect(waterline.testobjects.removeListItemsById)
+                    .to.be.calledWith(record.id, value);
+                });
+            });
+        });
+
+        describe('remove list items by bad id with removeListItemsById()',
+            function () {
+            it('should reject with a not found error', function () {
+                return waterline.testobjects.removeListItemsByIdentifier('invalid id', ["item1"])
+                .should.be.rejectedWith(Errors.NotFoundError);
+            });
+        });
+    });
+
+    describe('add item to list', function() {
+        var record;
+
+        before('reset DB collections', function () {
+            return helper.reset();
+        });
+
+        before('create the record', function () {
+            return waterline.testobjects.create({
+                identifiers: ["asd", "qwe"],
+                list: [{a: 1}]
+            })
+            .then(function (record_) {
+                record = record_;
+            });
+        });
+
+
+        describe('by Id with addListItemsIfNotExistById()', function () {
```

Comment:
I found the text of nested descript will be joined together in the unit test log.
So this describe + parent describe is a complete sentence.

Otherwise the sentence in log will be dupliated